src/implementations/twig/components/footer-standardised/footer-standardised.story.js

```diff
@@ -1,190 +1,140 @@
+  };
 
-  getExtraKnobs(data);
+  argTypes.hide_follow = {
+    name: 'follow us',
+    type: { name: 'boolean' },
+    defaultValue: false,
+    description: 'Hide "Follow us" section',
+    table: {
+      type: { summary: 'boolean' },
+      defaultValue: { summary: false },
+      category: 'States',
+    },
+  };
 
-  // Return the full specs.
-  return data;
+  if (system !== 'EU') {
+    argTypes.hide_about = {
+      name: 'about us',
+      type: { name: 'boolean' },
+      defaultValue: false,
+      description: 'Hide "About us" section',
+      table: {
+        type: { summary: 'boolean' },
+        defaultValue: { summary: false },
+        category: 'States',
+      },
+    };
+  }
+
+  argTypes.hide_relate_site = {
+    name: system === 'EU' ? 'optional links' : 'related sites',
+    type: { name: 'boolean' },
+    defaultValue: false,
+    description:
+      system === 'EU'
+        ? 'Hide "Optional links" section'
+        : 'Hide "Related sites" section',
+    table: {
+      type: { summary: 'boolean' },
+      defaultValue: { summary: false },
+      category: 'States',
+    },
+  };
+
+  if (system !== 'EU') {
+    argTypes.hide_class_name = {
+      name: 'class name',
+      type: { name: 'boolean' },
+      defaultValue: false,
+      description: 'hide "Class name" section',
+      table: {
+        type: { summary: 'boolean' },
+        defaultValue: { summary: false },
+        category: 'States',
+      },
+    };
+  }
+
+  return argTypes;
+};
+
+const prepareData = (data, args, system) => {
+  correctSvgPath(data);
+
+  if (data.rows[1][0][0].logo) {
+    data.rows[1][0][0].logo.src_mobile = logoEuMobile;
+    data.rows[1][0][0].logo.src_desktop = logoEuDesktop;
+  }
+
+  const res = JSON.parse(JSON.stringify(data));
+  if (args.hide_contact === true) {
+    res.rows[0][1].splice(0, 1);
+  }
+  if (args.hide_follow === true) {
+    res.rows[0][1].splice(1, 1);
+  }
+  if (args.hide_about === true) {
+    res.rows[0][2].splice(0, 1);
+  }
+  if (args.hide_relate_site === true) {
+    res.rows[0][2].splice(1, 1);
+  }
+  if (args.hide_class_name === true) {
+    res.rows.splice(1, 1);
+  }
+  if (args.hide_about === true && args.hide_relate_site === true) {
+    res.rows[0].splice(2, 1);
+  }
+  if (args.hide_relate_site === true && system === 'EU') {
+    res.rows[0].splice(2, 1);
+  }
+  if (args.hide_contact === true && args.hide_follow === true) {
+    res.rows[0].splice(1, 1);
+  }
+
+  return res;
 };
 
 export default {
   title: 'Components/Footers/Standardised',
-  decorators: [withCode, withNotes, withKnobs],
+  decorators: [withCode, withNotes],
+  parameters: {
+    knobs: { disable: true },
+  },
 };
 
-const data = process.env.STORYBOOK_SYSTEM === 'EU' ? specsEu : specsEc;
+const system = process.env.STORYBOOK_SYSTEM;
+const data = system === 'EU' ? specsEu : specsEc;
 
-export const Default = () => footer(formatFooter(data));
+export const Default = (args) => footer(prepareData(data, args, system));
```

Comment:
Why passing system to prepareData? system is a constant that you define in an higher scope, prepareData can use it, just to make it more readable and to avoid a linter complaint, put in on the top of the file, as we did in other stories.